pkg/hooks/hooks.go

```diff
@@ -82,14 +99,18 @@ func (h *hookContext) Close() error {
 //
 // NB: in the event of a timeout this will leak descriptors
 func (h *HookExecContext) RunWithTimeout(logger logging.Logger) error {
-	finished := make(chan struct{})
-	go func() {
-		h.Run(logger)
+	finished := make(chan RunHookResult)
+	go func(r chan RunHookResult) {
```

Comment:
I don't think you need to pass `finished` to this function. It could be a function that takes no params. You already referenced `finished` with the `close(finished)` line in the function.